plugins/org.python.pydev.parser/src/org/python/pydev/parser/prettyprinterv2/MakeAstValidForPrettyPrintingVisitor.java

```diff
@@ -1085,6 +1088,13 @@ public Object visitExpr(Expr node) throws Exception {
         return null;
     }
 
+    @Override
+    public Object visitExtSlice(ExtSlice node) throws Exception {
+        fixNode(node);
```

Comment:
The proper fix isn't to do the traverse but rather override it and then say what's required at each place.

So, you'd do code to iterate through `ExtSlice.dims` and then add `doc.addRequire(",", lastNode);` after each `dim` which you iterated (close to what the `org.python.pydev.parser.prettyprinterv2.PrettyPrinterVisitorV2.visitImport(Import)` does when handling the import names).

Reply:
Let me see if I could understand it.

So we should just return null with ExtSlice nodes (without traversing it nor using fixNode) and then visit it in PrettyPrinterVisitorV2, adding commas in between the dims?

(the exception throwed in this issue was about not having a ExtSlice visitor in MakeAstValidForPrettyPrintingVisitor)

Reply:
Yes (my comments there were related to the `PrettyPrinterVisitorV2.visitExtSlice`). 

On the `MakeAstValidForPrettyPrintingVisitor`, you also need to override `visitExtSlice`, but here I think it's something like:

```
    @Override
    public Object visitExtSlice(ExtSlice node) throws Exception {
        fixNode(node);
        if(node.dims != null) {
            for (int i = 0; i < node.dims.length; i++) {
                node.dims[i].accept(this);
                nextCol();
            }
        }
        fixAfterNode(node);
        return null;
    }
```